pkg/discovery/interface.go

```diff
@@ -26,11 +26,27 @@ type Discovery interface {
 	Type() string
 }
 
+// PluginDiscoveryCriteria provides criteria to look for plugins
+// in a discovery.
+type PluginDiscoveryCriteria struct {
+	// Name is the name of the plugin
+	Name string
+	// Target is the target of the plugin
+	Target configtypes.Target
+	// Version is the version for the plugin
+	Version string
+	// OS of the plugin binary in `GOOS` format.
+	OS string
+	// Arch of the plugin binary in `GOARCH` format.
+	Arch string
+}
+
 // CreateDiscoveryFromV1alpha1 creates discovery interface from v1alpha1 API
-func CreateDiscoveryFromV1alpha1(pd configtypes.PluginDiscovery) (Discovery, error) {
+func CreateDiscoveryFromV1alpha1(pd configtypes.PluginDiscovery, criteria *PluginDiscoveryCriteria) (Discovery, error) {
```

Comment:
I don't love adding the `criteria` here and not respecting it for any discovery other than OCI.
However, the go code using the discoveries does filter the plugin returned, so this is safe.